docs/documentation/recipes/static-website-with-mkdocs.md

```diff
@@ -0,0 +1,28 @@
+# How to create a static website for your documentation based on mkdocs and mkdocs-material
+
+[MKDocs](https://www.mkdocs.org/) is a tool built to create static websites from raw markdown files. Other alternatives include [Sphinx](https://www.sphinx-doc.org/en/master/), and [Jekyll](https://jekyllrb.com/).
+
+We used MKDocs to create [CSE Code-With Engineering Playbook](https://microsoft.github.io/code-with-engineering-playbook/) static website from the contents in [microsoft / code-with-engineering-playbook](https://github.com/microsoft/code-with-engineering-playbook) GitHub repository. Then we deployed it to a [GitHub Page](https://pages.github.com/).
+
+We found MKDocs to be a good choice since:
+
+1. It's easy to set up and looks great even with the vanilla version.
+2. It works well with markdown, which is what we already have in the Playbook.
+3. It uses a Python stack which is friendly to many contributors of this Playbook.
+
+For comparison, Sphinx mainly generates docs from restructured-text (rst) format, and Jekyll is written in Ruby.
+
+To setup an MKDocs website, the main assets needed are:
+
+1. An ```mkdocs.yaml``` file, similar to the one we have [in the Playbook](https://github.com/microsoft/code-with-engineering-playbook/blob/main/mkdocs.yml). This is the configuration file that defines the appearance of the website, the navigation, the plugins used and more.
+2. A folder named ```docs``` (the default value for the directory) that contains the documentation source files.
+3. A [Github Action](https://docs.github.com/actions/learn-github-actions/understanding-github-actions) for automatically generating the website (e.g. on every commit to main), similar to [this one from the Playbook](https://github.com/microsoft/code-with-engineering-playbook/blob/main/.github/workflows/mkdocs.yml).
+4. A list of plugins used during the build phase of the website. We specified ours [here](https://github.com/microsoft/code-with-engineering-playbook/blob/main/requirements-docs.txt). And these are the plugins we've used:
+
+    - [Material for MKDocs](https://squidfunk.github.io/mkdocs-material/): Material design appearance and user experience.
+    - [pymdown-extensions](https://facelessuser.github.io/pymdown-extensions/): Improves the appearance of markdown based content.
+    - [mdx_truly_sane_lists](https://github.com/radude/mdx_truly_sane_lists): For defining the indent level for lists without having to refactor the entire documentation we already had in the Playbook.
```

Comment:
Consider adding a link to other plugins that users might find useful, such as this one: https://github.com/mkdocs/mkdocs/wiki/MkDocs-Plugins and https://chrieke.medium.com/the-best-mkdocs-plugins-and-customizations-fc820eb19759

Reply:
Added those links. Thx!